### PR TITLE
Add support for sui product features

### PR DESCRIPTION
**Pull request by jvlcek**
https://bugzilla.redhat.com/show_bug.cgi?id=1637552

PR [16212](https://github.com/ManageIQ/manageiq/pull/16212) added support for
Service UI permissions.

This PR adds support for the API to honor the SUI permissions.

Chris Hale and I tested on a live appliance.

### CODE REVIEW

**Comment by jvlcek** [comment]
@chalettu Please review.

**Comment by jvlcek** [comment]
@miq-bot add_label bug
@miq-bot add_label gaprindashvili/yes
@miq-bot add_label hammer/yes

**Comment by jvlcek** [comment]
@miq-bot assign @gtanzillo

**Comment by jvlcek** [comment]
@gtanzillo Please review

**Comment by JPrause** [comment]
@miq-bot add_label blocker

**Comment by JPrause** [comment]
@gtanzillo with that approval,..can this be merged.

File: config/api.yml [hunk 1]
```diff
@@ -352,7 +352,10 @@
     :collection_actions:
       :get:
       - :name: read
-        :identifier: ops_settings
+        :identifier:
+        - ops_settings
+        - sui_services_show
```
Comment (gtanzillo) [inline config/api.yml]: @jvlcek, @chalettu Looking at this, it's not clear why a user should need to have `sui_services_show` and `sui_vm_details_view` to be able to request the `Category` collection. This change would no longer a user to request this collection when he previously could and the reason does not seems obvious.

File: config/api.yml [hunk 2]
```diff
@@ -365,7 +368,10 @@
     :resource_actions:
       :get:
       - :name: read
-        :identifier: ops_settings
+        :identifier:
+        - ops_settings
+        - sui_services_show
```
Comment (gtanzillo) [inline config/api.yml]: Same here

File: config/api.yml [hunk 3]
```diff
@@ -3194,7 +3340,10 @@
     :collection_actions:
       :get:
       - :name: read
-        :identifier: ops_settings
+        :identifier:
+        - ops_settings
+        - sui_services_show
```
Comment (gtanzillo) [inline config/api.yml]: And here, as well..

**Comment by JPrause** [comment]
@jvlcek @chalettu was there anything left before this can be merged. This is needed for a blocker.

**Comment by jvlcek** [comment]
@JPrause Yes, this is more work left to do. We've discussed with @gtanzillo and have identified a slight change that is in the works. It should be ready soon.

**Comment by gtanzillo** [comment]
Will merge once master is green

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit b1591eda3bffd89c5bea7da209550a7dc438f0f4
Author: Gregg Tanzillo <gtanzill@redhat.com>
Date:   Fri Nov 9 12:09:19 2018 -0500

    Merge pull request #501 from jvlcek/bz_1637552_apiyml
    
    Add support for sui product features
    
    (cherry picked from commit b7c13ca93d66f44eaf3a6b77ee9ffce046589246)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1637552
```

**Comment by simaishi** [comment]
@jvlcek There are multiple conflicts backporting to Gaprindashvili branch. Please create a separate PR. Thanks!

**Comment by simaishi** [comment]
Backported to Gaprindashvili via https://github.com/ManageIQ/manageiq-api/pull/510

**Pull request merged**